File: backend/src/main/java/com/app/backend/dtos/SubmitAnswerDTO.java
package com.app.backend.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class SubmitAnswerDTO {

    @NotNull
    @JsonProperty("tran_dau_id")
    private Long tranDauId;

    @NotNull
    @JsonProperty("cau_hoi_id")
    private Long cauHoiId;   // câu hiện tại

    @NotBlank
    @JsonProperty("answer")
    private String answer;   // A|B|C|D (case-insensitive)
}

File: frontend/src/app/dtos/tran-dau/submitanswer-dto.ts
export class SubmitAnswerDTO {
  tran_dau_id: number;
  cau_hoi_id: number;
  answer: string; // A/B/C/D

  constructor(tran_dau_id: number, cau_hoi_id: number, answer: string) {
    this.tran_dau_id = tran_dau_id;
    this.cau_hoi_id = cau_hoi_id;
    this.answer = answer;
  }
}
